### DISCUSSION TITLE
Welcome to MorseRunnerJS Discussions!

### DISCUSSION THREAD

**Discussion by mgiugliano** [mgiugliano/MorseRunnerJS#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by PierreLen** [comment]
Hello,

Where are you at for the dev ? i d like to participate, what upgrade can I work on ?

THX, F4JJZ

**Comment by mgiugliano** [comment]
Dear @PierreLen thanks for reaching out and for your interest!!

Since last year I did not have more time to continue beyond this "proof of concept". However I would be interested in your help, as CSS/JavaScript are not my strongest skills.
Something I see to be explored (in random order) would be

- polish and document the (bad) JS code I came out with
- explore whether Fabian Kurz's library is the best to simulate QSB (i.e. altering the volume, according to some slow random random process, over time?)
- reverse engineer and implement the original (FFT-based) method to generate a realistic RF-noise
- implement the QRM (annoying) stations invading the current frequency but not responding
- improve the management of "concurrent" random station popping out (when activity > 1)
- improve the concept of "distance", to simulate even better a response from the station those call sign is closest to the one actually sent by the user
- consider an integration with https://vail.woozle.org (I have been in touch with his creator)
- consider chatGPT-like engine for generating QSOs or rag-chews.. 
- use/adapt some CSS like (e.g.) https://botoxparty.github.io/XP.css/, for recreating the 
exact same experience of Morse Runner
- consider having a version that can be "played" through voice-chat and keyboard on a given Discord server: I have a half-cooked python script that does it.


These are random thoughts. I have no clear cut overall design yet and would love developing with your help, if you are interested.

All in all, I would love to increase (my own) opportunities to practice and learn CW when far from a radio or not feeling particularly brave.

73 de Mike iv3ifz
